File: python-sdk/src/boxsdk/models/operations/put_groups_id.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils
from ..shared import clienterror as shared_clienterror
from ..shared import group as shared_group


@dataclasses.dataclass
class PutGroupsIDPathParams:
    group_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'group_id', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class PutGroupsIDQueryParams:
    fields: Optional[list[str]] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'fields', 'style': 'form', 'explode': False }})
    
class PutGroupsIDRequestBodyInvitabilityLevelEnum(str, Enum):
    ADMINS_ONLY = "admins_only"
    ADMINS_AND_MEMBERS = "admins_and_members"
    ALL_MANAGED_USERS = "all_managed_users"

class PutGroupsIDRequestBodyMemberViewabilityLevelEnum(str, Enum):
    ADMINS_ONLY = "admins_only"
    ADMINS_AND_MEMBERS = "admins_and_members"
    ALL_MANAGED_USERS = "all_managed_users"


@dataclass_json
@dataclasses.dataclass
class PutGroupsIDRequestBody:
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    external_sync_identifier: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('external_sync_identifier') }})
    invitability_level: Optional[PutGroupsIDRequestBodyInvitabilityLevelEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invitability_level') }})
    member_viewability_level: Optional[PutGroupsIDRequestBodyMemberViewabilityLevelEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('member_viewability_level') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    provenance: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('provenance') }})
    

@dataclasses.dataclass
class PutGroupsIDRequest:
    path_params: PutGroupsIDPathParams = dataclasses.field()
    query_params: PutGroupsIDQueryParams = dataclasses.field()
    request: Optional[PutGroupsIDRequestBody] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PutGroupsIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    group: Optional[shared_group.Group] = dataclasses.field(default=None)
    

File: ts-sdk/src/sdk/models/operations/putgroupsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutGroupsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=group_id" })
  groupId: string;
}


export class PutGroupsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}

export enum PutGroupsIdRequestBodyInvitabilityLevelEnum {
    AdminsOnly = "admins_only",
    AdminsAndMembers = "admins_and_members",
    AllManagedUsers = "all_managed_users"
}

export enum PutGroupsIdRequestBodyMemberViewabilityLevelEnum {
    AdminsOnly = "admins_only",
    AdminsAndMembers = "admins_and_members",
    AllManagedUsers = "all_managed_users"
}


export class PutGroupsIdRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=external_sync_identifier" })
  externalSyncIdentifier?: string;

  @SpeakeasyMetadata({ data: "json, name=invitability_level" })
  invitabilityLevel?: PutGroupsIdRequestBodyInvitabilityLevelEnum;

  @SpeakeasyMetadata({ data: "json, name=member_viewability_level" })
  memberViewabilityLevel?: PutGroupsIdRequestBodyMemberViewabilityLevelEnum;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=provenance" })
  provenance?: string;
}


export class PutGroupsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutGroupsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: PutGroupsIdQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutGroupsIdRequestBody;
}


export class PutGroupsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  group?: shared.Group;

  @SpeakeasyMetadata()
  statusCode: number;
}
